### PR TITLE
Avoid divide by zero in superpmi.py tpdiff

### PR DESCRIPTION
**Pull request by BruceForstall**

### CODE REVIEW

**Comment by BruceForstall** [comment]
@jakobbotsch @kunalspathak @dotnet/jit-contrib PTAL

**Review by kunalspathak — APPROVED** [review]
LGTM

**Comment by JulieLeeMSFT** [comment]
@BruceForstall , are you going to backport this to 7 rc1?

**Review by jakobbotsch — APPROVED** [review]

**Comment by BruceForstall** [comment]
> @BruceForstall , are you going to backport this to 7 rc1?

no

File: src/coreclr/scripts/superpmi.py [hunk 1]
```diff
@@ -1956,9 +1956,12 @@ def replay_with_throughput_diff(self):
...
                     logging.info("Total instructions executed by diff: {}".format(diff_instructions))
-                    delta_instructions = diff_instructions - base_instructions
-                    logging.info("Total instructions executed delta: {} ({:.2%} of base)".format(delta_instructions, delta_instructions / base_instructions))
-                    tp_diffs.append((os.path.basename(mch_file), base_instructions, diff_instructions))
+                    if base_instructions != 0 and diff_instructions != 0:
+                        delta_instructions = diff_instructions - base_instructions
+                        logging.info("Total instructions executed delta: {} ({:.2%} of base)".format(delta_instructions, delta_instructions / base_instructions))
+                        tp_diffs.append((os.path.basename(mch_file), base_instructions, diff_instructions))
+                    else:
+                        logging.warning("One compilation failed to produce any results")
```
Comment (jakobbotsch) [inline src/coreclr/scripts/superpmi.py:1964]: Does asmdiffs need it too? When did we see this? Seems like things would need to be pretty catastrophically broken for it to happen.

Comment (BruceForstall) [reply src/coreclr/scripts/superpmi.py:1964]: I don't think asmdiffs needs it (I didn't see any failures), but I could be wrong. Yes, the "catastrophically broken" case is that currently one of the OSX-arm64 collections has a single MC, but it fails to replay due to "MISSING" error.

Comment (AndyAyersMS) [reply src/coreclr/scripts/superpmi.py:1964]: Can we just purge that single MC file? Or is it about to be overrwritten?

Comment (BruceForstall) [reply src/coreclr/scripts/superpmi.py:1964]: > Can we just purge that single MC file? Or is it about to be overrwritten?

I prefer to never do hand fixes to the Azure Storage data. And you're right, it should have been overridden this weekend, but that particular osx-arm64 job failed.

**Pull request merged**